src/amoc.erl

```diff
@@ -8,37 +8,49 @@
          add/1,
          remove/2]).
 
--export_type([
-              scenario/0,
-              do_opts/0,
-              remove_opts/0
-             ]).
-
+-export_type([scenario/0]).
 -type scenario() :: module().
 
--type do_opt() :: {nodes, [node()]} |
-                  {comment, string()} |
-                  {repeat, timeout()} |
-                  {interarrival, timeout()} |
-                  {config, amoc_config_scenario:settings()}.
--type do_opts() :: [do_opt()].
-
--type remove_opt() :: {force, boolean()}.
--type remove_opts() :: [remove_opt()].
-
 %% ------------------------------------------------------------------
 %% API for the local scenario execution, use amoc_dist module to run
 %% scenarios in a distributed environment
 %% ------------------------------------------------------------------
--spec do(amoc:scenario(), amoc_scenario:user_id(), amoc_scenario:user_id()) ->
+-spec do(scenario(), non_neg_integer(), amoc_config_scenario:config()) ->
     ok | {error, term()}.
-do(Scenario, Start, End) ->
-    amoc_controller:do(Scenario, Start, End).
-
--spec add(non_neg_integer()) -> ok.
-add(Count) ->
-    amoc_controller:add(Count).
-
--spec remove(non_neg_integer(), amoc:remove_opts()) -> ok.
-remove(Count, Opts) ->
-    amoc_controller:remove(Count, Opts).
\ No newline at end of file
+do(Scenario, Count, Settings) ->
+    case amoc_cluster:set_master_node(node()) of
+        ok ->
+            %% amoc_controller:start_scenario/2 will fail,
+            %% if amoc is running in a distributed mode
+            case {amoc_controller:start_scenario(Scenario, Settings), Count} of
+                {ok, 0} -> ok;
+                {ok, Count} -> amoc_controller:add_users(1, Count);
+                Error -> Error
+            end;
+        Error -> Error
+    end.
+
+-spec add(pos_integer()) -> ok | {error, any()}.
+add(Count) when is_integer(Count), Count > 0 ->
+    Node = node(),
+    case amoc_cluster:get_master_node() of
+        undefined -> {error, master_node_is_not_set};
+        Node ->
+            %% amoc_controller:add_users/2 will fail,
+            %% if amoc is running in a distributed mode
```

Comment:
it's a comment for the situation when someone manually connects to the slave clustered node (that is already running the test) and try to call amoc:do/3, amoc:add/2, amoc:remove/2 on the slave node.  instead of calling the proper amoc_dist interfaces from the master node